osu.Game/Rulesets/Ruleset.cs

```diff
@@ -25,13 +25,15 @@ public abstract class Ruleset
     {
         public readonly RulesetInfo RulesetInfo;
 
-        public IEnumerable<Mod> GetAllMods() => Enum.GetValues(typeof(ModType)).Cast<ModType>()
-                                                    // Confine all mods of each mod type into a single IEnumerable<Mod>
-                                                    .SelectMany(GetModsFor)
-                                                    // Filter out all null mods
-                                                    .Where(mod => mod != null)
-                                                    // Resolve MultiMods as their .Mods property
-                                                    .SelectMany(mod => (mod as MultiMod)?.Mods ?? new[] { mod });
+        public IEnumerable<Mod> GetAllMods(bool includeSystemMods) => Enum.GetValues(typeof(ModType)).Cast<ModType>()
```

Comment:
since `includeSystemMods` was set in 2 lines out of 5 (`APILegacyScoreInfo:L137` and `ScoreInfo:L76`), i think it would be better to set it as an optional parameter `bool includeSystemMods = false` (if we still want that parameter)